nf_core/modules/lint/main_nf.py

```diff
@@ -220,14 +220,18 @@ def check_process_section(self, lines):
             lspl = l.lstrip("https://").split(":")
             if len(lspl) == 2:
                 # e.g. 'https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img' :
-                singularity_tag = "_".join(lspl[0].split("/")[-1].strip().rstrip(".img").split("_")[1:])
+                if "YOUR-TOOL-HERE" not in lspl[0]:
+                    # Get singularity container version when a container is provided
+                    singularity_tag = re.search("biocontainers_(.*).img", lspl[0]).group(1)
```

Comment:
Literal dot and we want at least one character..

```suggestion
                    singularity_tag = re.search("biocontainers_(.+)\.img", lspl[0]).group(1)
```